### PR TITLE
FLN - 32 - Add swipe api

### PR DESCRIPTION
**Pull request by desolee**

### CODE REVIEW

File: flinder/flinderserver/api.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# 	"msg": "",
+# 	"data": []
+# }
+
+def get_match(request):
+    id = request.GET.get('id')
+    result = Swipe.objects.filter(swiper_id=id)
+    result_list = []
+    for title in result:
+        member = {"swiped_id": title.swiped_id, "swiper_id": title.swiper_id, "swipeRight": title.swipeRight}
```
Comment (space928) [inline flinder/flinderserver/api.py:19]: If the API could get the data in the form of:
```json
{
    "username":"hello@example.com",
    "photo":"media/images/example.png",
    "name":"Southpark House",
    "description":"Flat of 3 on Dumbarton Road"
}
```

File: flinder/flinderserver/api.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        result_list.append(member)
+
+    return JsonResponse({"status": "0", "msg": "", "data": result_list})
```
Comment (space928) [inline flinder/flinderserver/api.py:22]: Don't include unnecessary data in the json response. I don't need `status`, or `msg`, just return the result_list.

File: flinder/flinderserver/urls.py [hunk 3]
```diff
@@ -1,6 +1,7 @@
 from django.contrib import admin
 from django.urls import path, include
 from flinderserver import views
+from flinderserver.api import get_match
```
Comment (space928) [inline flinder/flinderserver/urls.py:4]: Import all of api.py, not individual functions.

**Review by space928 — CHANGES_REQUESTED** [review]
There's some stuff to change

File: flinder/flinderserver/api.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+# api/get_match?id=1
+# result example:
+# {
+# 	"status": "0",
+# 	"msg": "",
+# 	"data": []
+# }
+
+def get_match(request):
```
[the remark this answers is not in the document]
Comment (desolee) [reply flinder/flinderserver/api.py:14]: I think it's best if we leave it unchecked to make it easier to test

[the remark this answers is not in the document]
Comment (space928) [reply flinder/flinderserver/api.py:14]: It's fairly simple to test this with curl even with the request type set:
```bash
curl https://reqbin.com/echo/get/json -H "Accept: application/json"
```

[the remark this answers is not in the document]
Comment (desolee) [reply flinder/flinderserver/api.py:12]: Done

File: flinder/flinderserver/api.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    id = request.GET.get('id')
```
[the remark this answers is not in the document]
Comment (desolee) [reply flinder/flinderserver/api.py:15]: I haven't figured out how to include user in the request, is this the correct way?
image

**Review by space928 — APPROVED** [review]
It's still not perfect, but I'll merge so that we cna get on with other tasks.

**Pull request merged**